Whiskey/Functions/Invoke-WhiskeyDotNetCommand.ps1

```diff
@@ -0,0 +1,92 @@
+
+function Invoke-WhiskeyDotNetCommand
+{
+    <#
+    .SYNOPSIS
+    Runs `dotnet.exe` with a given SDK command and arguments.
+
+    .DESCRIPTION
+    The `Invoke-WhiskeyDotNetCommand` function runs the `dotnet.exe` executable with a given SDK command and any optional arguments. Pass the path to the `dotnet.exe` to the `DotNetPath` parameter. Pass the name of the SDK command to the `Name` parameter.
+
+    You may pass a list of arguments to the `dotnet.exe` command with the `ArgumentList` parameter. By default, the `dotnet.exe` command runs with any solution or .csproj files found in the current directory. To run the `dotnet.exe` command with a specific solution or .csproj file pass the path to that file to the `ProjectPath` parameter.
+
+    .EXAMPLE
+    Invoke-WhiskeyDotNetCommand -TaskContext $TaskContext -DotNetPath 'C:\Program Files\dotnet\dotnet.exe' -Name 'build' -ArgumentList '--verbosity minimal','--no-incremental' -ProjectPath 'C:\Build\DotNetCore.csproj'
+
+    Demonstrates running the following command `C:\> & "C:\Program Files\dotnet\dotnet.exe" build --verbosity minimal --no-incremental C:\Build\DotNetCore.csproj`
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [Whiskey.Context]
+        # The `Whiskey.Context` object for the task running the command.
+        $TaskContext,
+
+        [Parameter(Mandatory=$true)]
+        [string]
+        # The path to the `dotnet.exe` executable to run the SDK command with.
+        $DotNetPath,
+
+        [Parameter(Mandatory=$true)]
+        [string]
+        # The name of the .NET Core SDK command to run.
+        $Name,
+
+        [string[]]
+        # A list of arguments to pass to the .NET Core SDK command.
+        $ArgumentList,
+
+        [string]
+        # The path to a .NET Core solution or project file to pass to the .NET Core SDK command.
+        $ProjectPath
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    $dotNetExe = $DotNetPath | Resolve-Path -ErrorAction 'Ignore'
+    if (-not $dotNetExe)
+    {
+        Write-Error -Message ('"{0}" does not exist.' -f $DotNetPath)
+        return
+    }
+
+    $loggerArgs = & {

```

Comment:
These are MSBuild parameters and won't work on other dotnet commands. These should get moved up into the `DotNetBuild` task.

Reply:
I initially only had them for Build and Publish but then I tested these logging arguments with all the `DotNet` tasks and they worked, providing some relevant logging related to the dotnet command that was run. What it looked like was commands like `dotnet test` and `dotnet pack` were actually running MSBuild just with specific targets and so the logging args worked to create a log file.

I think [this documentation](https://docs.microsoft.com/en-us/dotnet/core/tools/cli-msbuild-architecture) confirms my suspicion of MSBuild running for all the `dotnet` commands.
>For example, the next version of Visual Studio will not call into dotnet restore (see note) command to restore dependencies for .NET Core projects, it will use the "Restore" target directly. Since these are MSBuild targets, you can also use raw MSBuild to execute them using the dotnet msbuild command.

With that in mind, should I keep the logging args for all `dotnet` commands or use them only for `DotNetBuild` and `DotNetPublish`?

Reply:
Leave them. I assumed `dotnet.exe` only called MSBuild when building.